### ISSUE TITLE
Signal handler test brittleness

### ISSUE THREAD

**Issue by GiudGiud** [idaholab/moose#24252]
## Bug Description
see https://civet.inl.gov/job/1500879/
seems the signal is sent too early

## Steps to Reproduce
Run test suite enough times

## Impact
Unexpected test failures

**Comment by lindsayad** [comment]
Seeing this on another PR in https://civet.inl.gov/job/1501716/

**Comment by lindsayad** [comment]
It sounds like @loganharbour may want to handle this a little differently in the future. Should we re-open?

**Comment by loganharbour** [comment]
Yes, this is still brittle... and still fails intermittently. I thought I made that clear on the PR, not sure why we closed

**Comment by lindsayad** [comment]
To add to this issue, these tests appear to not work at all with python 3.11

**Comment by lindsayad** [comment]
We still see this in CI. We had fixed the hypothetical case where MOOSE didn't wait long enough to receive the signal ... but we hadn't addressed the case where the test harness sends the signal before MOOSE is ready. You can trigger this pretty determinstically by setting HYDRA_DEBUG=1 in the environment such that early hydra stdout triggers the harness to send the signal before any MOOSE stdout
